packages/xrpl/src/models/transactions/oracleSet.ts

```diff
@@ -0,0 +1,189 @@
+import { ValidationError } from '../../errors'
+
+import {
+  BaseTransaction,
+  isNumber,
+  isString,
+  validateBaseTransaction,
+  validateOptionalField,
+  validateRequiredField,
+} from './common'
+
+const PRICE_DATA_SERIES_MAX_LENGTH = 10
+const SCALE_MAX = 10
+
+/**
+ * A PriceData object represents the price information for a token pair.
+ *
+ */
+export interface PriceData {
```

Comment:
This object should be used in the `Oracle` object as well.

Reply:
There's a type conflict with `AssetPrice`. It's `string` in `Oracle` but `number` in `OracleSet`. So I don't think it can be reused here.

Reply:
Done in https://github.com/XRPLF/xrpl.js/pull/2688/commits/845d811036d5c4f3086a3c3d668029c6f6aecd32